### DISCUSSION TITLE
Does ImageHeadersProvider  work for Compose UI?

### DISCUSSION THREAD

**Discussion by zeroarst** [GetStream/stream-chat-android#5271] [Q&A] [answered]
I asked how to add header in the avatar request earlier in this thread: https://github.com/GetStream/stream-chat-android/discussions/5267
Today I tried [ImageHeadersProvider](https://github.com/GetStream/stream-chat-android/blob/main/stream-chat-android-ui-components/src/main/kotlin/io/getstream/chat/android/ui/ChatUI.kt#L70) but it does now work. I noticed that it is coming from UI Component SDK. 
Does it work for Compose UI too? If no, what is the equavalent one in Compose?

**Comment by zeroarst** [comment]
I dug into the source code found that `Coli` is used for requesting avatar image, if I am not wrong.
That said, `Coli` does provide the ability to provide header or `okHttpClient` to the request https://coil-kt.github.io/coil/recipes/, but in the [`ImageUtils`](https://github.com/GetStream/stream-chat-android/blob/07276002eb7183c853da8ca4b9334d9bc7bd8fcb/stream-chat-android-compose/src/main/java/io/getstream/chat/android/compose/ui/util/ImageUtils.kt#L104), no interface is exposed so I guess currently there is no way to do that with Sream Compose SDK....

**Comment by JcMinarro** [comment]
Hello @zeroarst 
It is not available on Stream Compose SDK yet.
We have a task in our roadmap to provide it

**Comment by lachlan-eagling** [comment]
Hey @JcMinarro was there any timeline available for releasing this?

**Comment by JanePacey** [comment]
Hi @JcMinarro at the beginning of June we were advised this was scheduled for next sprint.  Do you have updates?  Thanks

**Comment by JcMinarro** [comment, marked as answer]
@JanePacey @lachlan-eagling @zeroarst We already implemented it on #5414 
It will be ready on the next release ;)
